### PR TITLE
Fiori 2

### PR DESCRIPTION
**Pull request by ukamo**

### CODE REVIEW

File: src/test/java/fiori/component/TabContainer.java [hunk 1]
```diff
@@ -21,4 +23,32 @@ public TabContainer(By locator) {
...
+     * @param tabName - the name of the tab to return
+     * @return WebElement tab
+     */
+    private WebElement getTab(String tabName) {
+        for (WebElement tab : getElement().findElements(By.className("th-bt-b"))) {
+            if (tab.getText().equals(tabName)) {
+                return tab;
+            }
+        }
+        return null;
```
Comment (nadya2712) [inline src/test/java/fiori/component/TabContainer.java:53]: It's better to return "throw new IllegalStateException("Cannot locate tab with label [" + tabName + "]");" instead null

File: src/test/java/fiori/component/TableCell.java [hunk 2]
```diff
@@ -23,4 +23,31 @@ protected TableCell(WebElement element) {
     public Hyperlink getHyperlink() {
         return functionPassed(() -> new Hyperlink(getElement().findElement(By.className("th-tx-value"))));
     }
+
+    /**
+     * The method gets the hyperlink by checkbox from the table cell.
+     *
+     * @return Hyperlink
+     */
+    public Hyperlink getHyperlinkByCheckbox() {
```
Comment (nadya2712) [inline src/test/java/fiori/component/TableCell.java]: What does it mean getHyperlinkByCheckbox(). Please, change method name to getHyperlink()

File: src/test/java/fiori/component/TableRow.java [hunk 3]
```diff
@@ -34,15 +34,43 @@ public TableCell getCell(int columnIndex) {
...
+     *
+     * @param columnIndex - the column order, starts with 0.
+     * @return TableCell
+     */
+    public TableCell getCellItem(int columnIndex) {
+        return functionPassed(() -> {
+            if (columnIndex >= table.getHeaders().size()) {
+                throw new IllegalStateException("Cannot find column index [" + columnIndex + "] in the table.");
+            }
+            return new TableCell(getElement().findElements(By.className("th-clr-td")).get(columnIndex));
```
Comment (nadya2712) [inline src/test/java/fiori/component/TableRow.java:34]: Are you sure the locator By.className("th-clr-td") fits to all cells in table?

File: src/test/java/fiori/screens/CommonDialog.java [hunk 4]
```diff
@@ -26,4 +31,16 @@ public int getRowsNumberFromTable(String dialogTitle) {
     public void clickHyperlinkByNameInTableColumn(String dialogTitle, String hyperlinkValue, String columnName) {
         getDialogByTitle(dialogTitle).getTable().getColumn(columnName).getRow(hyperlinkValue).getCell(columnName).getHyperlink().click();
     }
+
+    public void selectRowWithCheckedCheckbox(String columnHeader, String dialogTitle) {
+        List<WebElement> rows = getDialogByTitle(dialogTitle).getTable().getRows();
+        functionPassed(() -> {
+            for (WebElement row : rows) {
+                if (getDialogByTitle(dialogTitle).getTable().getRow(rows.indexOf(row)).getCell(columnHeader).getCheckbox().isChecked()) {
+                    getDialogByTitle(dialogTitle).getTable().getRow(rows.indexOf(row)).getCell(columnHeader).getHyperlinkByCheckbox().click();
```
Comment (nadya2712) [inline src/test/java/fiori/screens/CommonDialog.java]: Сode should be implementation in component classes. For example:
getDialogByTitle(dialogTitle).getTable().getRowWithCheckedCheckbox(column).select()

File: src/test/resources/features/fiori/Fiori.feature [hunk 5]
```diff
@@ -30,17 +30,25 @@ Feature: Release New Service Order Quotation in application Fiori
...
-    #When I switch to tab Items on screen Service Order Quotations in application Fiori
-    #Then I assert I am on tab Items on screen Service Order Quotations in application Fiori
-    #And I assert table is displayed on tab Items on screen Service Order Quotations in application Fiori
-    #When I set Product ID "BUND01" in row "1" in the table on tab Items on screen Service Order Quotations in application Fiori
-    #And I set Quantity "1" in row "1" in the table on tab Items on screen Service Order Quotations in application Fiori
-    #And I check Select unchecked checkboxes in the table on tab Items on screen Service Order Quotations in application Fiori
-    #Then I assert checkboxes are checked in the table on tab Items on screen Service Order Quotations in application Fiori
+    When I select row with checked checkbox in Default column in table on dialog Partner Selection Document header on screen Service Order Quotations in application Fiori
+    Then I assert dialog Partner Selection Document header is closed on screen Service Order Quotations in application Fiori
+    And I assert I am on screen Service Order Quotations in application Fiori
```
Comment (nadya2712) [inline src/test/resources/features/fiori/Fiori.feature]: Excess step

File: src/test/java/fiori/steps/ServiceOrderQuotationSteps.java [hunk 6]
```diff
@@ -22,4 +24,34 @@ public void setInputByLabelOnTab(String labelName, String value, String tab, Str
...
+        serviceOrderQuotationScreen.switchToTab(tabName);
+    }
+
+    @When("^I check checkbox in column (Select) in table on tab (Items) on screen (Service Order Quotations) in application (Fiori)$")
+    public void checkCheckboxesInColumn(String columnHeader, String tabName, String screen, String app) {
+        serviceOrderQuotationScreen.checkCheckboxesInColumn(columnHeader);
+    }
+
+    @Then("^I click button (Save) on tab (Items) on screen (Service Order Quotations) in application (Fiori)$")
+    public void clickSaveButton(String buttonName, String tabName, String screen, String app) {
```
Comment (nadya2712) [inline src/test/java/fiori/steps/ServiceOrderQuotationSteps.java]: should be clickButton()

File: src/test/java/fiori/steps/ServiceOrderQuotationSteps.java [hunk 7]
```diff
@@ -22,4 +24,34 @@ public void setInputByLabelOnTab(String labelName, String value, String tab, Str
...
+        if (columnHeader.equals("Quantity")) {
+            serviceOrderQuotationScreen.setInputInRowItemsTableAndClickEnter(columnHeader, value, rowIndex - 1);
+        } else {
+            serviceOrderQuotationScreen.setInputInRowItemsTable(columnHeader, value, rowIndex - 1);
+
+        }
+    }
+
+    @Then("^I assert table is displayed on tab (Items) on screen (Service Order Quotations) in application (Fiori)$")
+    public void assertItemsTableIsDisplayed(String tabName, String screen, String app) {
```
Comment (nadya2712) [inline src/test/java/fiori/steps/ServiceOrderQuotationSteps.java]: Should be isTableDisplayed

File: src/test/java/fiori/screens/ServiceOrderQuotationScreen.java [hunk 8]
```diff
@@ -18,4 +28,37 @@ public void setInputByLabelAndClickEnter(String labelName, String value) {
...
+        input.sendKeys(Keys.ENTER);
+    }
+
+    public void checkCheckboxesInColumn(String columnHeader) {
+        List<WebElement> rows = itemsTable.getRows();
+        Wait.functionPassed(() -> {
+            rows.forEach((row) -> {
+                if (!itemsTable.getRow(rows.indexOf(row)).getCell(columnHeader).getElement().findElements(By.className("th-sapcb-a")).isEmpty()) {
+                    itemsTable.getRow(rows.indexOf(row)).getCell(columnHeader).getCheckbox().check();
+                }
```
Comment (nadya2712) [inline src/test/java/fiori/screens/ServiceOrderQuotationScreen.java]: Сode should be implementation in component classes. For example:
 public void checkAllCheckboxes(String column) {
        table.checkAllCheckBoxes(column);
    }
The same for method above

File: src/test/java/fiori/screens/ServiceOrderQuotationScreen.java [hunk 9]
```diff
@@ -18,4 +28,37 @@ public void setInputByLabelAndClickEnter(String labelName, String value) {
...
+    }
+
+    public void setInputInRowItemsTable(String columnHeader, String value, int rowIndex) {
+        itemsTable.getRow(rowIndex).getCell(columnHeader).getInput().setText(value);
+    }
+
+    public void setInputInRowItemsTableAndClickEnter(String columnHeader, String value, int rowIndex) {
+        Input input = itemsTable.getRow(rowIndex).getCell(columnHeader).getInput();
+        input.setText(value);
+        input.sendKeys(Keys.ENTER);
```
Comment (nadya2712) [inline src/test/java/fiori/screens/ServiceOrderQuotationScreen.java:53]: Maybe better to create method setTextWithEnter(value) inTableCell class

File: src/test/java/fiori/component/TableRow.java (hunk 3 above)
Comment (ukamo) [reply src/test/java/fiori/component/TableRow.java:34]: yes, each td has class "th-clr-td"

File: src/test/java/fiori/steps/HomeSteps.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    public void clickHyperlink(String hyperlinkName, String screen, String app) {
+        homeScreen.switchToTab(hyperlinkName);
+    }
+
+    /*@When("^I switch to tab (Manage Solution Quotations) on screen (Home) in application (Fiori)$")
+    public void switchToTab(String tabName, String screen, String app)  {
+        homeScreen.switchToTabTile(tabName);
+    }*/
+
+    @And("^I click hyperlink (Manage Solution Quotations) on screen (Home) in application (Fiori)$")
```
Comment (nadya2712) [inline src/test/java/fiori/steps/HomeSteps.java:21]: When

File: src/test/resources/features/fiori/Fiori.feature [hunk 11]
```diff
@@ -30,17 +30,27 @@ Feature: Release New Service Order Quotation in application Fiori
...
+    Then I assert I am on tab Items on screen Service Order Quotations in application Fiori
+    And I assert table is displayed on tab Items on screen Service Order Quotations in application Fiori
+    When I set Product ID "BUND01" in row 1 in table on tab Items on screen Service Order Quotations in application Fiori
+    And I set Quantity "1" in row 1 in table on tab Items on screen Service Order Quotations in application Fiori
+    And I check checkbox in column Select in table on tab Items on screen Service Order Quotations in application Fiori
+    And I click button Save on tab Items on screen Service Order Quotations in application Fiori
+    Then I assert icon with checkbox is presented on screen Service Order Quotations in application Fiori
+    When I click icon with checkbox on screen Service Order Quotations in application Fiori
+    And I store Transaction number on screen Service Order Quotations in application Fiori
+    When I click icon Back on screen Service Order Quotations in application Fiori
```
Comment (nadya2712) [inline src/test/resources/features/fiori/Fiori.feature]: And

File: src/test/resources/features/fiori/Fiori.feature [hunk 12]
```diff
@@ -30,17 +30,27 @@ Feature: Release New Service Order Quotation in application Fiori
...
+    Then I assert I am on screen Home in application Fiori
+    When I switch to tab Subscription Management on screen Home in application Fiori
+    And I click hyperlink Manage Solution Quotations on screen Home in application Fiori
+    Then I assert I am on screen Service Order Quotations in application Fiori
+    When I set Transaction number on Service Order Quotation ID on screen Service Order Quotations in application Fiori
+    When I click button Search on screen Service Order Quotations in application Fiori
```
Comment (nadya2712) [inline src/test/resources/features/fiori/Fiori.feature]: And

**Pull request merged**